### PR TITLE
CR-1109011: load xclbin errors in sysfs

### PR DESCRIPTION
**Pull request by sarab96**
Add support for load xclbin errors into sysfs node

### CODE REVIEW

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_ioctl.c [hunk 1]
```diff
@@ -354,10 +355,18 @@ static int xocl_preserve_mem(struct xocl_drm *drm_p, struct mem_topology *new_to
 
 static bool xocl_xclbin_in_use(struct xocl_dev *xdev)
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_ioctl.c:356]: xocl_xclbin_in_use() is a simple function that performs the check, the side effect of inserting an error into some global list is not expected. please move them to outside of this function.

**Review by maxzhen — CHANGES_REQUESTED** [review]

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_ioctl.c (hunk 1 above)
Comment (sarab96) [reply src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_ioctl.c]: updated

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_ioctl.c [hunk 2]
```diff
@@ -377,20 +378,36 @@ xocl_read_axlf_helper(struct xocl_drm *drm_p, struct drm_xocl_axlf *axlf_ptr)
...
 
+
+	err_last.pid = 0;
+	err_last.ts = 0; //TODO timestamp
 	if (!xocl_is_unified(xdev)) {
 		userpf_err(xdev, "XOCL: not unified Shell\n");
+		err_last.err_code = XRT_ERROR_CODE_BUILD(XRT_ERROR_NUM_XCLBIN_INCOMPATIBLE, 
+			XRT_ERROR_DRIVER_XOCL, XRT_ERROR_SEVERITY_CRITICAL, 
+			XRT_ERROR_MODULE_XCLBIN, XRT_ERROR_CLASS_SYSTEM);
+		xocl_insert_error_record(&xdev->core, &err_last);
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_ioctl.c:391]: I think this is for a not supported shell, not for incompatible xclbin. @houlz0507 can point you to a more suitable location where xclbin compatibility is checked.

Comment (houlz0507) [reply src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_ioctl.c:391]: all the xclbin checks on xocl side are in xocl_read_axlf_helper().

**Review by larry9523 — COMMENTED** [review]
So we are using the error handling frame work which was built to handle asynchronous error to handle synchronous error too? 

If so, we may want to update our error reporting conf page as well so that we have a clear document about what synchronous error are reported through xrtErrorGetLast

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_ioctl.c (hunk 2 above)
Comment (sarab96) [reply src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_ioctl.c]: updated

**Comment by sarab96** [comment]
Yes I will ask for that doc/info to be updated

> So we are using the error handling frame work which was built to handle asynchronous error to handle synchronous error too?
> 
> If so, we may want to update our error reporting conf page as well so that we have a clear document about what synchronous error are reported through xrtErrorGetLast

**Comment by maxzhen** [comment]
> 
> 
> Yes I will ask for that doc/info to be updated
> 
> > So we are using the error handling frame work which was built to handle asynchronous error to handle synchronous error too?
> > If so, we may want to update our error reporting conf page as well so that we have a clear document about what synchronous error are reported through xrtErrorGetLast

I'm not sure if this is desired. We have return code, do we need a 2nd way to get these errors? If this is not in Larry's original design, we need to discuss more. I've marked this PR as do-not-merge until we have a clear design first.

**Comment by sarab96** [comment]
Will add other async erros

**Pull request closed without merging**